Retry on every exception.

jumarko opened the pull request:
Catching `ExceptionInfo` is not enough and the whole `auto` task will
fail if we don't exception `Exception`.
By catching exception, we can let `lein auto` keep trying indefinitely,
even when the task is failing.

jumarko committed:
Retry on all exceptions

Catching ExceptionInfo is not enough and the whole lein auto task
will fail if we don't catch Exception. By catching Exception,
we can let lein auto keep trying indefinitely,
even when an underlying task is failing.

weavejester commented:
Thanks for the patch. What exception did you get that broke it?

Can you also remove the version number change and change the commit message to follow the [seven rules](https://chris.beams.io/posts/git-commit/#seven-rules).

jumarko commented:
@weavejester I was trying to use it for `lein antlr`. When there's a problem in the ANTLR grammar, I got an error and the whole `lein auto antlr` process exits:
```
auto> Running: lein antlr
Compiling ANTLR grammars: GoComplexity.g4 ScalaMethodArguments.g4 PLSqlSharedRulesLexer.g4 Swift.g4 ObjectiveCSharedRulesLexer.g4 CppMethodArguments.g4 SwiftNestedComplexity.g4 GroovySharedRulesLexer.g4 CppSharedRulesLexer.g4 Kotlin.g4 JavaNestedComplexity.g4 Go.g4 CSharpCohesion.g4 ObjectiveCMethodArguments.g4 JavaMicro.g4 ObjectiveCSharedRulesParser.g4 VisualBasicComplexity.g4 TSql.g4 GroovySharedRulesParser.g4 Clojure.g4 RubyMicro.g4 PHPNestedComplexity.g4 PLSql.g4 JavaComplexity.g4 JavaCohesion.g4 Scala.g4 Groovy.g4 SwiftMethodArguments.g4 SwiftCohesion.g4 VisualBasicNestedComplexity.g4 CStyleNestedComplexity.g4 PLSqlComplexity.g4 PHP.g4 ScalaSharedRulesLexer.g4 ObjectiveCMicro.g4 CSharpSharedRulesLexer.g4 GoNestedComplexity.g4 VisualBasic.g4 PHPComplexity.g4 CSharpComplexity.g4 GroovyNestedComplexity.g4 SwiftSharedRulesLexer.g4 ScalaSharedRulesParser.g4 CSharpSharedRulesParser.g4 ObjectiveCNestedComplexity.g4 Erlang.g4 TSqlSharedRulesLexer.g4 ScalaComplexity.g4 CStyleComplexity.g4 JavaSharedRulesLexer.g4 CppSharedRulesParser.g4 JavaMethodArguments.g4 ECMAScriptNestedComplexity.g4 CppMicro.g4 CSharpMicro.g4 ECMAScript.g4 Apex.g4 Python3.g4 CSharpMethodArguments.g4 PHPSharedRulesLexer.g4 GoMethodArguments.g4 VisualBasicSharedRulesLexer.g4 JavaSharedRulesParser.g4 ScalaNestedComplexity.g4 GroovyMethodArguments.g4 KotlinComplexity.g4 ECMAScriptComplexity.g4 GoSharedRulesLexer.g4 CSharpNestedComplexity.g4 SwiftComplexity.g4 ...
error(50): TSql.g4:69:0: syntax error: unterminated rule (missing ';') detected at 'create_or_alter_function :' while looking for rule element
java.lang.RuntimeException: ANTLR detected 1 grammar errors.
 at leiningen.antlr$process_antlr_dir.invokeStatic (antlr.clj:141)
    leiningen.antlr$process_antlr_dir.invoke (antlr.clj:124)
    leiningen.antlr$compile_antlr.invokeStatic (antlr.clj:153)
    leiningen.antlr$compile_antlr.invoke (antlr.clj:143)
    leiningen.antlr$antlr.invokeStatic (antlr.clj:179)
    leiningen.antlr$antlr.invoke (antlr.clj:164)
    clojure.lang.Var.invoke (Var.java:379)
    clojure.lang.AFn.applyToHelper (AFn.java:154)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.core$apply.invokeStatic (core.clj:648)
    clojure.core$apply.invoke (core.clj:641)
    leiningen.core.main$partial_task$fn__4667.doInvoke (main.clj:284)
    clojure.lang.RestFn.invoke (RestFn.java:410)
    clojure.lang.AFn.applyToHelper (AFn.java:154)
    clojure.lang.RestFn.applyTo (RestFn.java:132)
    clojure.lang.AFunction$1.doInvoke (AFunction.java:29)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invokeStatic (core.clj:648)
    clojure.core$apply.invoke (core.clj:641)
    leiningen.core.main$apply_task.invokeStatic (main.clj:334)
    leiningen.core.main$apply_task.invoke (main.clj:320)
    leiningen.core.main$resolve_and_apply.invokeStatic (main.clj:340)
    leiningen.core.main$resolve_and_apply.invoke (main.clj:336)
    leiningen.auto$run_task.invokeStatic (auto.clj:37)
    leiningen.auto$run_task.invoke (auto.clj:35)
    leiningen.auto$auto$fn__559.invoke (auto.clj:81)
    leiningen.auto$auto.invokeStatic (auto.clj:80)
    leiningen.auto$auto.doInvoke (auto.clj:64)
    clojure.lang.RestFn.invoke (RestFn.java:425)
    clojure.lang.Var.invoke (Var.java:383)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.core$apply.invokeStatic (core.clj:648)
    clojure.core$apply.invoke (core.clj:641)
    leiningen.core.main$partial_task$fn__4667.doInvoke (main.clj:284)
    clojure.lang.RestFn.applyTo (RestFn.java:139)
    clojure.lang.AFunction$1.doInvoke (AFunction.java:29)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invokeStatic (core.clj:648)
    clojure.core$apply.invoke (core.clj:641)
    leiningen.core.main$apply_task.invokeStatic (main.clj:334)
    leiningen.core.main$apply_task.invoke (main.clj:320)
    leiningen.core.main$resolve_and_apply.invokeStatic (main.clj:340)
    leiningen.core.main$resolve_and_apply.invoke (main.clj:336)
    leiningen.core.main$_main$fn__4734.invoke (main.clj:420)
    leiningen.core.main$_main.invokeStatic (main.clj:411)
    leiningen.core.main$_main.doInvoke (main.clj:408)
    clojure.lang.RestFn.invoke (RestFn.java:421)
    clojure.lang.Var.invoke (Var.java:383)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.core$apply.invokeStatic (core.clj:646)
    clojure.main$main_opt.invokeStatic (main.clj:314)
    clojure.main$main_opt.invoke (main.clj:310)
    clojure.main$main.invokeStatic (main.clj:421)
    clojure.main$main.doInvoke (main.clj:384)
    clojure.lang.RestFn.invoke (RestFn.java:457)
    clojure.lang.Var.invoke (Var.java:394)
    clojure.lang.AFn.applyToHelper (AFn.java:165)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.main.main (main.java:37)

```

I've also updated the commit message - not sure to which problems you were referring, but I removed the trailing comma in the subject line.

weavejester commented:
Thanks, this looks fine. Can you remove the backticks from the commit message and put the two sentences in a single paragraph?

jumarko commented:
Done.

weavejester commented:
I fixed the line width and manually merged in the PR.

jumarko commented:
James, thanks for your fix and sorry for that.
I'm not used to following such strict commit message rules so it's easy to screw it up O:-).
